algo.js

```diff
@@ -441,7 +461,7 @@ const allMatches = async () => {
   })
 }
 
-const test = async (title, prevDocs) => {
+const test = async (title, jsonObj) => {
```

Comment:
Correct me if I'm wrong but it seems like the test() function is almost identical to the actual main function. There's a lot of duplicate code and we have to update both of them each time we want to make a change to one of them. Also, the whole point of test function is that it simulates the main function, but if they're written separately there's room for error if we don't copy the code properly. So maybe we should combine them into one function but put if statement's throughout the function that run the proper mode based on the user input of test vs. run.

Reply:
yea i was actually thinking the same thing before